app/src/main/java/com/geekskool/dhobi/Activities/AddCourseActivity.java

```diff
@@ -0,0 +1,110 @@
+package com.geekskool.dhobi.Activities;
+
+import android.content.ContentValues;
+import android.os.Bundle;
+import android.support.annotation.Nullable;
+import android.support.design.widget.Snackbar;
+import android.support.v7.app.AppCompatActivity;
+import android.view.View;
+import android.widget.DatePicker;
+import android.widget.EditText;
+
+import com.geekskool.dhobi.Contracts.Contract.CourseEntry;
+import com.geekskool.dhobi.Models.Course;
+import com.geekskool.dhobi.R;
+
+import java.util.Calendar;
+
+import static com.geekskool.dhobi.Contracts.Contract.CourseEntry.COLUMN_DURATION;
+import static com.geekskool.dhobi.Contracts.Contract.CourseEntry.COLUMN_END_DATE;
+import static com.geekskool.dhobi.Contracts.Contract.CourseEntry.COLUMN_LOCATION;
+import static com.geekskool.dhobi.Contracts.Contract.CourseEntry.COLUMN_NAME;
+import static com.geekskool.dhobi.Contracts.Contract.CourseEntry.COLUMN_START_DATE;
+import static com.geekskool.dhobi.Util.isInvalid;
+
+/**
+ * Created by manisharana on 20/2/17.
+ */
+public class AddCourseActivity extends AppCompatActivity implements DatePicker.OnDateChangedListener {
+
+    private EditText mName;
+    private EditText mLocation;
+    private EditText mDuration;
+    private DatePicker mStartDate;
+    private long startDate;
+    private View rootView;
+    private final static int MINUTE = 60000;
+
+
+    @Override
+    protected void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_add_course);
+        rootView = findViewById(android.R.id.content);
+        Calendar cal = Calendar.getInstance();
+        int year = cal.get(Calendar.YEAR);
+        int month = cal.get(Calendar.MONTH);
+        int day = cal.get(Calendar.DAY_OF_MONTH);
+
+        mName = (EditText) findViewById(R.id.et_course_name);
+        mDuration = (EditText) findViewById(R.id.et_course_duration);
+        mLocation = (EditText) findViewById(R.id.et_course_location);
+        mStartDate = (DatePicker) findViewById(R.id.dp_start_date);
+        mStartDate.init(year, month, day, this);
+        mStartDate.setMinDate(cal.getTimeInMillis() - MINUTE);
+    }
+
+    public void closeDialogBox(View view) {
+        finish();
+    }
+
+
+    public void validateCourse(View view) {
+        Course course = getCourse();
+
+        if (course == null)
+            Snackbar.make(rootView, R.string.should_not_be_empty, Snackbar.LENGTH_LONG).show();
+        else {
+            saveCourse(course);
+            finish();
+        }
+
+    }
+
+    private Course getCourse() {
+        String name = mName.getText().toString();
+        if (isInvalid(name)) return null;
+        String duration = mDuration.getText().toString();
+        if (isInvalid(duration)) return null;
+        String location = mLocation.getText().toString();
+        if (isInvalid(location)) return null;
```

Comment:
The above 6 lines look like validation, which would preferably happen outside of the View layer. Manisha and Priyanka missed this initial talk about design/architecture and I can discuss it again quickly, if you'd like.